src/overlays/actors/ovl_En_Go/z_en_go.c

```diff
@@ -1222,286 +1625,373 @@ s32 func_80A13B1C(EnGo* this, PlayState* play) {
+        this->actor.child = EnGo_FindGravemaker(this, play);
         this->actor.child->child = &this->actor;
-        func_80A141D4(this, play);
-        this->actionFunc = func_80A14E14;
+        EnGo_ChangeToFrozenAnimation(this, play);
+        this->actionFunc = EnGo_Frozen;
     } else {
-        func_80A143A8(this, play);
-        this->actionFunc = func_80A149B0;
+        EnGo_ChangeToShiveringAnimation(this, play);
+        this->actionFunc = EnGo_Idle;
     }
 }
 
-void func_80A14668(EnGo* this, PlayState* play) {
-    if (!(gSaveContext.save.weekEventReg[22] & 4)) {
-        func_80A14324(this, play);
-        this->actionFunc = func_80A149B0;
+/**
+ * Setup a shrine Goron.
+ *
+ * There are three, but they all behave the same way aside from dialog.
+ */
+void EnGo_ShrineGoron_Setup(EnGo* this, PlayState* play) {
+    if (!(gSaveContext.save.weekEventReg[22] & 0x4)) { // Elder's son has been calmed
+        EnGo_ChangeToCoveringEarsAnimation(this, play);
+        this->actionFunc = EnGo_Idle;
     } else {
-        func_80A143A8(this, play);
-        this->actionFunc = func_80A149B0;
+        EnGo_ChangeToShiveringAnimation(this, play);
+        this->actionFunc = EnGo_Idle;
     }
 }
 
-void func_80A146CC(EnGo* this, PlayState* play) {
-    func_80A134B0(this, play, 0);
-    func_80A12C48(this, play, 0);
-    this->unk_3A4 *= 5.0f;
-    Actor_SetScale(&this->actor, this->unk_3A4);
+/**
+ * Setup the Medigoron in the Powder Keg Shop.
+ *
+ * This Goron sells Powder Kegs, and gives the player a Powder Keg as part of a challenge to gain
+ * the privilege to buy them.
+ */
+void EnGo_Medigoron_Setup(EnGo* this, PlayState* play) {
+    EnGo_Medigoron_UpdateColliderRadius(this, play, false);
+    EnGo_ChangeAnim(this, play, ENGO_ANIM_LYINGDOWNIDLE);
+    this->scaleFactor *= ENGO_MEDIGORON_SCALE_MULTIPLIER;
+    Actor_SetScale(&this->actor, this->scaleFactor);
     this->actor.flags &= ~ACTOR_FLAG_1;
     this->actor.targetMode = 3;
-    this->unk_390 = 0;
+    this->actionFlags = ENGO_FLAG_NONE;
     this->actor.gravity = -1.0f;
-    SubS_UpdateFlags(&this->unk_390, 3, 7);
-    this->unk_390 |= 0x40;
-    this->unk_390 |= 0x20;
-    this->unk_3D8 = func_80A13E80;
-    this->actionFunc = func_80A149B0;
-}
-
-void func_80A14798(EnGo* this, PlayState* play) {
-    EffectTireMarkInit sp38 = {
-        0,
-        62,
-        { 0, 0, 15, 100 },
-    };
+    SubS_UpdateFlags(&this->actionFlags, 3, 7);
+    this->actionFlags |= ENGO_FLAG_LOST_ATTENTION;
+    this->actionFlags |= ENGO_FLAG_EYES_OPEN;
+    this->msgEventFunc = EnGo_Medigoron_CutsceneGivePowderKeg;
+    this->actionFunc = EnGo_Idle;
+}
 
-    if ((this->unk_288 < 0) || SubS_IsObjectLoaded(this->unk_288, play) || (this->unk_289 < 0) ||
-        SubS_IsObjectLoaded(this->unk_289, play)) {
+/**
+ * Default setup function as the initial action function.
+ */
+void EnGo_SetupAction(EnGo* this, PlayState* play) {
```

Comment:
Think it would make more sense to call this SetupInitialAction, then, SetupAction we usually use for a function that changes the action whenever it changes.